portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -89,28 +90,31 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       
         Task task = new Task(id, fname, lname, timeStamp, postedComment);
         tasks.add(task);
+
     }
+
         Gson gson = new Gson();
 
+        // Creates a sublist of the original array list
+        List<Task> maxCommentTask = tasks.subList(0, Math.min(tasks.size(), setNumOfComments(request)));
         response.setContentType("application/json;");
-        response.getWriter().println(gson.toJson(tasks));
-
-        setNumOfComments(request);
-
+        String toJson = gson.toJson(maxCommentTask);
+        //System.out.println("HERE! " + toJson);
+        response.getWriter().println(toJson);
 
     }
-
+        //Parses the string from the queryString to a int
          private int setNumOfComments(HttpServletRequest request) {
 
         //Get the user input from form
-        String userCommentLimit = request.getParameter("number-of-comments");
+        String maxCommentsStr= request.getParameter("maxComments");
```

Comment:
for future reference, it's better to store these constant strings into a constant variable. For exmample
`private static final String PARAM_COMMENT_NUM = "number-of-comments`
And use PARAM_COMMENT_NUM here